File: facturator-client/src/app/shared/models/article.ts
export class Article{
  id: any = "";
  name: string = "";
  photo: string = "";
  price: number = 0;
  description: string = "";

  constructor(_name:string, price:number, _photo:string, description:string, _id?:string ) {
    this.id = _id;
    this.name = _name;
    this.photo = _photo;
    this.price = price;
    this.description = description;
  }
}

File: facturator-api-dotnetcore/facturator-api/Models/Article.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace facturator_api.Models
{
    public class Article
    {
        public Article(string name, string photoUrl, decimal price, string description)
        {
            this.Name = name;
            this.PhotoUrl = photoUrl;
            this.Price = price;
            this.Description = description;
        }
        public Article(string name, string photoUrl, decimal price, string description, int id) : this(name, photoUrl, price, description)
        {
            this.Id = id;
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public string PhotoUrl { get; set; }
        public decimal Price { get; set; }
        public string Description{ get; set; }
    }
}
